### ISSUE TITLE
[Bug] rpc-file moved upon migration

### ISSUE THREAD

**Issue by NicolasDorier** [ElementsProject/lightning#3378]
So this one is weird.
I ran my tests on 0.8 => no problem.
But when I try deploying on mainnet, it seems the `rpc-file` parameter is completely ignored.

The only difference between my tests and the mainnet is that in mainnet, the data folder has been migrated from /root/.lightning/ to /root/.lightning/bitcoin, on regtest, there was no migration.

On mainnet my `/root/.lightning/config` is

```
bitcoin-datadir=/etc/bitcoin
bitcoin-rpcconnect=bitcoind

proxy=tor:9050

bind-addr=0.0.0.0:9735
network=bitcoin
announce-addr=nico2.temp.btcpayserver.org:9735
announce-addr=uewlcjmuuwcgs43lpcc7d64k2pf4xs7yyer7ztsq6cblyjxue3bc35qd.onion:9735
rpc-file=/root/.lightning/lightning-rpc
```

On regtest (where rpc-file works) it is

```
bitcoin-datadir=/etc/bitcoin
bitcoin-rpcconnect=bitcoind
network=regtest
announce-addr=lightningd
log-level=debug

bind-addr=0.0.0.0:9735
network=regtest
rpc-file=/root/.lightning/lightning-rpc
```

But on mainnet the rpc-file still show up in the bitcoin folder.

Do you see anything I would have missed?

Edit: in the log I saw:

```
2019-12-27T13:55:42.579Z UNUSUAL lightningd: Moving lightning-rpc into /root/.lightning/bitcoin/
```

During the migration. It does not appear again if I restart. But it should not have been moved anyway.

**Comment by darosior** [comment]
> on regtest, there was no migration.

Weird, should be `.lightning/regtest/`.

> Do you see anything I would have missed?
> Edit: in the log I see:

In 0.8 the `lightning_dir` was separated into networks, you shouldn't be able to access the RPC socket at the root of the `lightning_dir` anymore.

**Comment by NicolasDorier** [comment]
@darosior on regtest there is no problem. The lightning-rpc file is created in .lightning and the data is in /regtest. 

But on mainnet, the data is in /bitcoin as well as the lightning-rpc.

**Comment by darosior** [comment]
You should prefer using `lightning-dir=` (superfluous as you seem to be using the default) and `network=`, I think.

**Comment by darosior** [comment]
> But on mainnet, the data is in /bitcoin as well as the lightning-rpc.

Ok I just checked my node and indeed, if the `network` is `bitcoin` then the socket is moved too. I'm not sure that was intended..

**Comment by NicolasDorier** [comment]
@darosior I really want the data directory to be in /bitcoin so the user don't mix up his data with testnet. However, for backward compatibility reason I don't want to change the lightning-rpc file location.

**Comment by NicolasDorier** [comment]
@darosior if the problem was that the socket is moved, I would have expect that to be fixed on restart, but that does not seem to be the case.

**Comment by darosior** [comment]
It won't be fixed on restart as the migration is only done once. Btw that could help you to patch this, just copy the socket after the migration it won't be moved anymore (detection is done with `hsm_secret`)

**Comment by NicolasDorier** [comment]
@darosior I am quite surprised by it.
I thought the socket was recreated everytimes the app restart.
How can the app know the socket is in `bitcoin/` when it restart?

**Comment by darosior** [comment]
> How can the app know the socket is in bitcoin/ when it restart?

Using the `network` config option, which now defaults to `bitcoin`.

**Comment by NicolasDorier** [comment]
But in this case why does in regtest it works when there is no migration?

**Comment by NicolasDorier** [comment]
Ok so I tried to removed my datadir and restart again. The problem does not happen. So this is a bug in the migration.

**Comment by NicolasDorier** [comment]
Closing this.
I tried to reproduce but I could not.

The lightning-rpc file is moved when it should not be, but it actually does not have any side effect.
The lightning-rpc file seems to be still created in `.lightning` folder as well as expected.

In my first test it did not, because, I assume, I was restoring an old node which could not catched up the sync because blocks were pruned, and so the process was not listening on lightning-rpc .
